From 5f596aa0ee1ff1d5afe34bbbea8de9c0a790d51b Mon Sep 17 00:00:00 2001
From: Marius Storm-Olsen <marius.storm-olsen@nokia.com>
Date: Tue, 22 Mar 2011 16:12:33 -0500
Subject: [PATCH] QtWebKit/Windows: Link with Advapi32

OSRandomSource.cpp uses
    CryptAcquireContext
    CryptGenRandom
    CryptReleaseContext
which requires symbols from the Advapi32 library on Windows.
---
 Source/JavaScriptCore/jsc.pro |    3 +++
 Source/WebCore/WebCore.pri    |    4 +---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/Source/JavaScriptCore/jsc.pro b/Source/JavaScriptCore/jsc.pro
index a550c48..f265700 100644
--- a/Source/JavaScriptCore/jsc.pro
+++ b/Source/JavaScriptCore/jsc.pro
@@ -31,6 +31,9 @@ wince* {
     LIBS += mmtimer.lib
 }
 
+# for APIs used in wtf/OSRandomSource.cpp
+win32*:LIBS += -ladvapi32
+
 # Prevent warnings about difference in visibility on Mac OS X
 contains(QT_CONFIG, reduce_exports):CONFIG += hide_symbols
 unix:contains(QT_CONFIG, reduce_relocations):CONFIG += bsymbolic_functions
diff --git a/Source/WebCore/WebCore.pri b/Source/WebCore/WebCore.pri
index ba902f7..9cd4a8f 100644
--- a/Source/WebCore/WebCore.pri
+++ b/Source/WebCore/WebCore.pri
@@ -210,11 +210,8 @@ contains(DEFINES, ENABLE_NETSCAPE_PLUGIN_API=1) {
     }
     win32-* {
         LIBS += \
-            -ladvapi32 \
-            -lgdi32 \
             -lshell32 \
             -lshlwapi \
-            -luser32 \
             -lversion
     }
 }
@@ -306,6 +303,7 @@ contains (CONFIG, text_breaking_with_icu) {
 }
 
 win32-* {
+    LIBS += -ladvapi32
     LIBS += -lgdi32
     LIBS += -lole32
     LIBS += -luser32
-- 
1.7.2.3.msysgit.0.311.gb27be

